s3_upload/utils/log.py

```diff
@@ -35,29 +39,81 @@ def set_file_handler(logger, log_dir) -> logging.Logger:
     logging.Logger
         logging handler
     """
-    log_file = os.path.join(log_dir, "s3_upload.log")
+    check_write_permission_to_log_dir(log_dir)
+
+    log_file = os.path.join(
+        log_dir, f"s3_upload.log.{datetime.now().strftime('%Y-%m-%d')}"
+    )
+
+    existing_file_handler = [
+        x for x in logger.handlers if isinstance(x, FileHandler)
+    ]
 
-    if any(
-        [isinstance(x, TimedRotatingFileHandler) for x in logger.handlers]
-    ) and os.path.exists(log_file):
-        # log file handler already set and log file exists => use it
-        logger.info("Log file handler already set to %s", log_file)
+    if existing_file_handler and os.path.exists(log_file):
+        logger.debug("FileHandler already set")
         return logger
 
+    Path(log_dir).mkdir(parents=True, exist_ok=True)
+    Path(log_file).touch(exist_ok=True)
+
+    file_handler = FileHandler(filename=log_file)
+    file_handler.setFormatter(FORMATTER)
+
+    logger.addHandler(file_handler)
+
     logger.info(
-        "Initialised log fileHandler, setting log output to %s", log_file
+        "Initialised log FileHandler, setting log output to %s", log_file
     )
 
-    check_write_permission_to_log_dir(log_dir)
+    clear_old_logs(logger=logger, log_dir=log_dir, backup_count=5)
+
+    return logger
+
+
+def clear_old_logs(logger, log_dir, backup_count) -> None:
```

Comment:
what? it's there already
<!-- Reviewable comment -ODG-wTMAKE7dxOcMHF-:bpaml25 -->
<!-- Sent from Reviewable.io -->

Reply:
what the rabbit said, but it seems redundant since its in the docstring and we aren't generating any auto docs etc from this, so ignoring
<!-- Reviewable comment -ODG6CPY49kmxHTaagba:b5gkuct -->
<!-- Sent from Reviewable.io -->

Reply:
ah it didn't suggest a fix for the type in the parameters passed but in the docstring which was already there
<!-- Reviewable comment -ODGALRe0Ous5k2-hfnT:b-39ugnn -->
<!-- Sent from Reviewable.io -->

Reply:
Done.
<!-- Reviewable comment -ODGDjsV8rzc_8AP48B6:b-896fix -->
<!-- Sent from Reviewable.io -->